ctx.go

```diff
@@ -423,11 +426,22 @@ func body[B any](c ContextNoBody) (B, error) {
 	case "application/x-www-form-urlencoded", "multipart/form-data":
 		body, err = readURLEncoded[B](c.Req, c.readOptions)
 	case "application/xml":
-		return readXML[B](c.Req.Context(), c.Req.Body, c.readOptions)
+		body, err = readXML[B](c.Req.Context(), c.Req.Body, c.readOptions)
 	case "application/x-yaml":
-		return readYAML[B](c.Req.Context(), c.Req.Body, c.readOptions)
+		body, err = readYAML[B](c.Req.Context(), c.Req.Body, c.readOptions)
 	case "application/json":
 		fallthrough
```

Comment:
Is this not a bug now? Wouldn't the fallthrough hit the next case?

Like [this playground](https://go.dev/play/p/0DxTOo7PbuX)

Reply:
There is no fallthrough for yaml ! Only for JSON, to allow the default case to be JSON

Reply:
Yeah we don't have a test where we explicitly declare `application/json` content header. Which is why we're missing this.

Reply:
Oh yes you're right, I thought you were talking about the lines above. We definitely need a test for this case so it never happens again